internal/dag/builder.go

```diff
@@ -388,7 +399,9 @@ func (b *Builder) compute() *DAG {
 		dag.roots = append(dag.roots, vh)
 	}
 	for _, svh := range _svhosts {
-		dag.roots = append(dag.roots, svh)
+		if svh.secret != nil {
```

Comment:
Since Root & Delegate objects are combined, the only way to know if someone specifies TLS on a Root is to look for the existence of the secret on the object.